src/main/java/com/leahtcodes/Controllers/CoffeeController.java

```diff
@@ -42,13 +50,23 @@ record NewCoffeeRequest(
             Integer coffeePrice,
             String roast
     ){}
+
     @PostMapping("/add")
-    public String addCoffee(@RequestBody NewCoffeeRequest request){
+    public String addCoffee(@RequestBody NewCoffeeRequest request, OAuth2AuthenticationToken authentication){
+        OAuth2AuthorizedClient client = authorizedClientService
+                .loadAuthorizedClient(
+                        authentication.getAuthorizedClientRegistrationId(),
+                        authentication.getName());
+        System.out.println(authentication.getPrincipal().getAttributes());
```

Comment:
I know we have this right now for debugging, but good idea to remove this in prod code UNLESS deploying to the cloud. 

Also, you're calling the .getAttributes on a method you can't guarantee is present. Look at this https://spring.io/guides/topicals/spring-security-architecture/ and that provides some documentation around how to use an Authenticated person/user.  This is great start!